tools/codegen/api/types.py

```diff
@@ -145,7 +152,10 @@ def name(self) -> str:
     # Render the C++ declaration for this signature
     def decl(self) -> str:
         returns_type = cpp.returns_type(self.func.returns)
-        cpp_args_str = ', '.join(a.decl() for a in self.arguments())
+        cpp_args_str = ', '.join(
+            a.decl() if a.name not in self.cpp_no_default_args else a.no_default().decl()
+            for a in self.arguments()
```

Comment:
Curious, why are you keeping the defaults in `cpp.arguments` when they don't exist in the API? 

Also, don't think this interacts with `TensorOptions` at all. So, if that creates ambiguity then there's no way to resolve it.

Reply:
I think you're right, should push this in more. It's kind of annoying if you push this all the way into `cpp.argument` though, since then it becomes pretty wordy.